test/e2e/launch.e2e.test.js

```diff
@@ -1,108 +1,84 @@
 const path = require('path')
 const fs = require('fs-extra')
 const tmp = require('tmp')
-const delay = require('delay')
-const chai = require('chai')
-const dirtyChai = require('dirty-chai')
 const { makeRepository } = require('./utils/ipfsd')
 const portfinder = require('portfinder')
 
-const expect = chai.expect
-chai.use(dirtyChai)
-
-// To print the app logs, add the following to your test:
-//
-// const logs = await app.client.getMainProcessLogs()
-// logs.forEach(line => console.log(line))
-//
 async function getPort () {
   return portfinder.getPortPromise()
 }
 
-// Note: logs before "start daemon FINISHED" event are consumed
-// inside of `daemonReady`. To print them, pass DEBUG=true
-
-describe('Application launch', function () {
-  this.timeout(60000)
+test.describe.serial('Application launch', async () => {
   let app = null
 
-  afterEach(async function () {
-    if (app && app.isRunning()) {
-      await app.stop()
+  test.afterEach(async () => {
+    if (app) {
+      await app.close()
     }
   })
 
-  async function startApp ({
-    home = tmp.dirSync({ prefix: 'tmp_HOME_', unsafeCleanup: true }).name,
-    repoPath = path.join(home, '.ipfs')
-  }) {
-    app = new Application({
-      path: electronPath,
-      args: ['-r', path.join(__dirname, 'utils/include.js'), path.join(__dirname, '../../src/index.js')],
-      env: {
+  async function startApp ({ repoPath } = {}) {
+    const home = tmp.dirSync({ prefix: 'tmp_home_', unsafeCleanup: true }).name
+    if (!repoPath) {
+      repoPath = path.join(home, '.ipfs')
+    }
+    app = await electron.launch({
+      args: [path.join(__dirname, '../../src/index.js')],
+      env: Object.assign({}, process.env, {
         NODE_ENV: 'test',
         HOME: home,
         IPFS_PATH: repoPath
-      }
+      })
     })
-    await app.start()
     return { app, repoPath, home }
   }
 
-  async function daemonReady (app, timeout = 45000) {
-    // TODO: replace this hack with a signal from the app to know when ipfs is ready.
-    // Right now we can't listen for IPC events in spectron (https://github.com/electron/spectron/issues/91)
-    // As a workaround, we look at console output and match on strings :<
-    const tick = 250
-    const ready = (output) => output && output.match(/(?:daemon is running|Daemon is ready|start daemon FINISHED)/)
-    const hasPeerId = (output) => output && output.trim().match(/PeerID is (\w+)$/)
-    let peerId
-    while (true) {
-      const logs = await app.client.getMainProcessLogs()
-      for (const line of logs) {
-        if (process.env.DEBUG) console.log(line)
-        const idMatch = hasPeerId(line)
-        if (idMatch) peerId = idMatch[1]
-        if (ready(line)) {
-          return { peerId }
+  async function daemonReady (app) {
+    const peerId = await app.evaluate(async ({ ipcMain }) => new Promise((resolve, reject) => {
+      ipcMain.on('ipfsd', (status, peerId) => {
```

Comment:
:information_source: really cool, switching to more sane way of reading ipfsd status makes tests run really fast now!